From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: opencode <opencode@localhost>
Date: Sun, 2 Aug 2026 00:00:00 +0000
Subject: [PATCH] use Android ABI padding for POSIX_MQUEUE user_struct fields

Enable CONFIG_POSIX_MQUEUE without breaking kernel ABI on the ZTE UMS9620
(MiFi F50) kernel 5.4.210. ZTE's android_kabi.h does not define
ANDROID_OEM_DATA_ARRAY, so only ANDROID_KABI_USE/RESERVE are used.

Signed-off-by: opencode <opencode@localhost>
---
 include/linux/sched/user.h | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/include/linux/sched/user.h b/include/linux/sched/user.h
index 6d63a5260130..b96fe706b5ef 100644
--- a/include/linux/sched/user.h
+++ b/include/linux/sched/user.h
@@ -23,7 +23,7 @@
 #endif
 #ifdef CONFIG_POSIX_MQUEUE
 	/* protected by mq_lock	*/
-	unsigned long mq_bytes;	/* How many bytes can be allocated to mqueue? */
+//	unsigned long mq_bytes;	/* How many bytes can be allocated to mqueue? */
 #endif
 	unsigned long locked_shm; /* How many pages of mlocked shm ? */
 	unsigned long unix_inflight;	/* How many files in flight in unix sockets */
@@ -41,8 +41,13 @@
 	/* Miscellaneous per-user rate limit */
 	struct ratelimit_state ratelimit;
 
+#if defined(CONFIG_POSIX_MQUEUE)
+	ANDROID_KABI_USE(1, unsigned long mq_bytes);
+	ANDROID_KABI_RESERVE(2);
+#else
 	ANDROID_KABI_RESERVE(1);
 	ANDROID_KABI_RESERVE(2);
+#endif
 };
 
 extern int uids_sysfs_init(void);
